python/tvm/script/parser/core/entry.py

```diff
@@ -77,4 +95,15 @@ def parse(program: Union[doc.AST, Any, str], extra_vars: Dict[str, Any] = None)
             parser.parse(extra_vars=extra_vars)
         except ParserError as err:
             parser.report_error(err.node, err.args[0])
-    return builder.get()
+    ret = builder.get()
+    # well-formedness check will ignore any non-Relax functions
```

Comment:
Instead of ignoring the non-Relax functions, can we run the TIR equivalent `tvm.tir.analysis.verify_well_formed` function?

Reply:
Long-term, I think it would be good to have a single validation pass, rather than the IR-specific passes, because an `IRModule` may contain both TIR and Relax functions.

Reply:
Will do, I was not aware that there was a similar analysis in TIR. We should definitely apply both.